lib/tether/ops_linux.go

```diff
@@ -784,23 +784,58 @@ func (t *BaseOperations) Cleanup() error {
 	return nil
 }
 
-// Need to put this here because Windows does not
-// support SysProcAttr.Credential
-func getUserSysProcAttr(uname string) *syscall.SysProcAttr {
-	uinfo, err := user.Lookup(uname)
+// Need to put this here because Windows does not support SysProcAttr.Credential
+// getUserSysProcAttr relies on docker user package to verify user specification
+// Examples of valid user specifications are:
+//     * ""
+//     * "user"
+//     * "uid"
+//     * "user:group"
+//     * "uid:gid
+//     * "user:gid"
+//     * "uid:group"
+func getUserSysProcAttr(uid, gid string) (*syscall.SysProcAttr, error) {
+	if len(uid) == 0 && len(gid) == 0 {
+		log.Debugf("no user id or group id specified")
+		return nil, nil
+	}
+
+	user := uid
+	if len(gid) > 0 {
+		user = fmt.Sprintf("%s:%s", uid, gid)
+	}
+	execUser, err := searchUser(user)
 	if err != nil {
-		detail := fmt.Sprintf("Unable to find user %s: %s", uname, err)
-		log.Error(detail)
-		return nil
+		return nil, err
 	}
-	u, _ := strconv.Atoi(uinfo.Uid)
-	g, _ := strconv.Atoi(uinfo.Gid)
-	// Unfortunately lookup GID by name is currently unsupported in Go.
-	return &syscall.SysProcAttr{
+
+	sysProc := &syscall.SysProcAttr{
 		Credential: &syscall.Credential{
-			Uid: uint32(u),
-			Gid: uint32(g),
+			Uid: uint32(execUser.Uid),
+			Gid: uint32(execUser.Gid),
 		},
 		Setsid: true,
 	}
+	for _, sgid := range execUser.Sgids {
+		sysProc.Credential.Groups = append(sysProc.Credential.Groups, uint32(sgid))
+	}
+	return sysProc, nil
+}
+
+func searchUser(user string) (*dockerUser.ExecUser, error) {
```

Comment:
This doesn't seem significant enough to warrant splitting out into a separate function if we have to pack/unpack data to return it. This could be inlined or return the Credentials struct directly.